drivers/ethernet/eth_nxp_enet.c

```diff
@@ -155,6 +166,86 @@ static void net_if_mcast_cb(struct net_if *iface,
 }
 #endif /* CONFIG_NET_NATIVE_IPV4 || CONFIG_NET_NATIVE_IPV6 */
 
+/*
+ *****************
+ * PTP Functions *
+ *****************
+ */
+
+#if defined(CONFIG_PTP_CLOCK_NXP_ENET)
+static bool eth_get_ptp_data(struct net_if *iface, struct net_pkt *pkt)
+{
+	int eth_hlen;
+
+#if defined(CONFIG_NET_VLAN)
+	struct net_eth_vlan_hdr *hdr_vlan;
+	struct ethernet_context *eth_ctx;
+	bool vlan_enabled = false;
+
+	eth_ctx = net_if_l2_data(iface);
+	if (net_eth_is_vlan_enabled(eth_ctx, iface)) {
+		hdr_vlan = (struct net_eth_vlan_hdr *)NET_ETH_HDR(pkt);
+		vlan_enabled = true;
+
+		if (ntohs(hdr_vlan->type) != NET_ETH_PTYPE_PTP) {
+			return false;
+		}
+
+		eth_hlen = sizeof(struct net_eth_vlan_hdr);
+	} else
+#endif /* CONFIG_NET_VLAN */
+	{
+		if (ntohs(NET_ETH_HDR(pkt)->type) != NET_ETH_PTYPE_PTP) {
+			return false;
+		}
+
+		eth_hlen = sizeof(struct net_eth_hdr);
+	}
+
+	net_pkt_set_priority(pkt, NET_PRIORITY_CA);
+
+	return true;
+}
+
+
+#if defined(CONFIG_NET_L2_PTP)
+static inline void ts_register_tx_event(const struct device *dev,
+					 enet_frame_info_t *frameinfo)
+{
+	struct nxp_enet_mac_data *data = dev->data;
+	struct net_pkt *pkt;
+
+	pkt = frameinfo->context;
+	if (pkt && atomic_get(&pkt->atomic_ref) > 0) {
```

Comment:
Should the minimum ref count be actually 2?

If the reference count == 1, then the packet will be freed by net_pkt_unref() on line 235. That means whatever we push into the queue by net_if_add_tx_timestamp() is no longer valid.